### PR TITLE
Chapter 20 web api rest

### PR DESCRIPTION
**Pull request by jwoolbright23**

### CODE REVIEW

File: content/web-api-rest/reading/web-api/_index.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+##### JSON
+
+JSON is currently a universal representation of data accepted by client applications. This means our web API packages data 
+requested by a client application as JSON. The web API also transfers this JSON in its communication with a client application.
+
+<!-- TODO: Link will need to be changed from localhost once we have a permanent endpoint -->
+Let's revisit the last two steps from our [web API work flow](http://localhost:1313/web-api-rest/web-api/#transference-of-data)
+
+5. The controller transforms the object into a JSON representation.
+6. The controller responds to the client with the JSON representation.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/web-api/_index.md:132]:
```suggestion
1. The controller transforms the object into a JSON representation.
1. The controller responds to the client with the JSON representation.
```

File: content/web-api-rest/reading/web-api/_index.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+{{% notice green "Tip" "rocket" %}}
+[XML](https://developer.mozilla.org/en-US/docs/Web/XML/XML_introduction) is another popular data format. It is now used less commonly than 
+JSON for web API-to-client communications.
+{{% /notice %}}
+
+In the next section, we will discuss exactly how a client application makes a request and how a web API responds.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/web-api/_index.md:139]: Is this referring to the section below? Might reword:
```suggestion
Let us discuss exactly how a client application makes a request and how a web API responds.
```

File: content/web-api-rest/exercises/_index.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+You can then issue a `GET` request to the `Location` header value and view the new entity. 
+
+If the request fails because of a client error, then it will respond with a `400` status code and a message about what went wrong. In the case of 
+`CodingEvent` entities, the following validation criteria must be met:
+
+- `Title`: 10-100 characters
+- `Description`: less than 1000 characters
+ 
+{{% notice blue "Question" "rocket" %}}
```
Comment (codinglikeagirl42) [inline content/web-api-rest/exercises/_index.md:180]:
```suggestion
{{% notice green "Question" "rocket" %}}
```

File: content/web-api-rest/reading/web-api/_index.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+   "title": "An Astronaut's Guide to Life on Earth",
+   "author": "Chris Hadfield",
+   "ISBN": 9780316253017,
+   "year_published": 2013,
+   "subject": ["Hadfield, Chris", "Astronauts", "Biography"],
+   "available": true
+}
+```
+
+The HTTP body may include JSON that represents the data being passed between an API and client application. Remember, not all requests/responses include HTTP bodies.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/web-api/_index.md:192]:
```suggestion
An HTTP body can contain a large number of different media types, 
known as [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types). A MIME type is associated with the HTTP header `Content-Type`. This header instructs the recipient of the HTTP request/response on what MIME type the HTTP body contains. We've seen a `Content-Type:text/html` HTTP header before. 

{{% notice green "Example" "rocket" %}}
```html {linenos=table}
<!DOCTYPE html>
<html>
   <head>
      <title>My Web Page</title>
      content
   </head>
   <body>
      content
   </body>
</html>
```

This is the header for HTML documents and is used throughout the web. APIs send representations of data in the format of JSON requiring the header `Content-Type` to be `application/json`. This allows us to pass the state of the data as the HTTP body.

```json {linenos=table}
{
   "title": "An Astronaut's Guide to Life on Earth",
   "author": "Chris Hadfield",
   "ISBN": 9780316253017,
   "year_published": 2013,
   "subject": ["Hadfield, Chris", "Astronauts", "Biography"],
   "available": true
}
```

The HTTP body may include JSON that represents the data being passed between an API and client application. Remember, not all requests/responses include HTTP bodies.
{{% notice %}}
```

File: content/web-api-rest/reading/what-is-rest/_index.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    "Id": 1,
+    "Title": "Halloween Hackathon!",
+    "Description": "A gathering of nerdy ghouls...",
+    "Date": "2020-10-31"
+    },
+    ...
+]
+```
+
+Notice that the state here is represented as the collective state of all of the ``CodingEvents`` in the collection.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/what-is-rest/_index.md:83]:
```suggestion
{{% notice green "Example" "rocket" %}}
The state of a *single* ``CodingEvent`` entity is represented as a single JSON object:

```json {linenos=table}
{
    "Id": 1,
    "Title": "Halloween Hackathon!",
    "Description": "A gathering of nerdy ghouls...",
    "Date": "2020-10-31"
}
```

Whereas the state of a collection of `CodingEvents` is represented by a JSON array of objects.

```json {linenos=table}
[
    {
    "Id": 1,
    "Title": "Halloween Hackathon!",
    "Description": "A gathering of nerdy ghouls...",
    "Date": "2020-10-31"
    },
    ...
]
```

Notice that the state here is represented as the collective state of all of the ``CodingEvents`` in the collection.
{{% /notice %}}
```

File: content/web-api-rest/reading/what-is-rest/_index.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+{{% /notice %}}
+
+### What is a Representation?
+
+Representation refers to a depiction of state that is usable in a given context. In the REST context, state must be 
+represented in a way that is portable and compatible with both the client and API. 
+
+JSON is a data format that provides structure, portability and compatibility. For these reasons, JSON is the standard representation used when transferring state between a client application and an API. 
+
+{{% notice blue "Example" "rocket" %}}
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/what-is-rest/_index.md:56]:
```suggestion
{{% notice green "Example" "rocket" %}}
```

File: content/web-api-rest/reading/what-is-rest/_index.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+The state of a *single* ``CodingEvent`` entity is represented as a single JSON object:
+{{% /notice %}}
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/what-is-rest/_index.md:58]:
```suggestion
```

File: content/web-api-rest/reading/what-is-rest/_index.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    "Title": "Halloween Hackathon!",
+    "Description": "A gathering of nerdy ghouls...",
+    "Date": "2020-10-31"
+    },
+    ...
+]
+```
+
+Notice that the state here is represented as the collective state of all of the ``CodingEvents`` in the collection.
+
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/what-is-rest/_index.md:84]:
```suggestion
{{% /notice %}}

```

Comment (codinglikeagirl42) [reply content/web-api-rest/reading/what-is-rest/_index.md:84]: These code blocks should be include in the green Example block.

File: content/web-api-rest/reading/web-api/_index.md [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+
+We call HTTP a *stateless protocol*. State can be a complex concept that refers to a number of things. We'll explore some aspects of it in more depth on the next page. 
+
+In the context of HTTP, think of **state** as information about application data that is transferred via HTTP bodies, HTTP status codes, and HTTP headers.
+
+#### Bodies
+
+An HTTP body can contain a large number of different media types, 
+known as [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types). A MIME type is associated with the HTTP header `Content-Type`. This header instructs the recipient of the HTTP request/response on what MIME type the HTTP body contains. We've seen a `Content-Type:text/html` HTTP header before. Here's a simple example of it:
+
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/web-api/_index.md:165]:
```suggestion
known as [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types). A MIME type is associated with the HTTP header `Content-Type`. This header instructs the recipient of the HTTP request/response on what MIME type the HTTP body contains. We've seen a `Content-Type:text/html` HTTP header before. 

{{% notice blue "Example" "rocket" %}}
```

File: content/web-api-rest/reading/web-api/_index.md [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+   "author": "Chris Hadfield",
+   "ISBN": 9780316253017,
+   "year_published": 2013,
+   "subject": ["Hadfield, Chris", "Astronauts", "Biography"],
+   "available": true
+}
+```
+
+The HTTP body may include JSON that represents the data being passed between an API and client application. Remember, not all requests/responses include HTTP bodies.
+
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/web-api/_index.md:193]:
```suggestion
{{% notice %}}

```

Comment (codinglikeagirl42) [reply content/web-api-rest/reading/web-api/_index.md:193]: Put example in blue example box.

File: content/web-api-rest/reading/rest-fundamentals/_index.md [hunk 11]
```diff
@@ -0,0 +1,444 @@
...
+{{% notice blue "Example" "rocket" %}}
+```csharp {linenos=table}
+public class CodingEvent {
+    public int Id { get; set; }
+    public string Title { get; set; }
+    public string Description { get; set; }
+    public DateTime Date { get; set; }
+}
+```
+{{% /notice %}}
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/rest-fundamentals/_index.md:43]:
```suggestion
```

File: content/web-api-rest/reading/rest-fundamentals/_index.md [hunk 12]
```diff
@@ -0,0 +1,444 @@
...
+The output resource shape of a CodingEvent entity:
+
+```bash {linenos=table}
+CodingEvent {
+    Id: integer
+    Title: string
+    Description: string
+    Date: string (ISO 8601 date format)
+}
+```
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/rest-fundamentals/_index.md:56]:
```suggestion
```
{{% /notice %}}
```

File: content/web-api-rest/reading/rest-fundamentals/_index.md [hunk 13]
```diff
@@ -0,0 +1,444 @@
...
+
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/rest-fundamentals/_index.md:57]:
```suggestion
{{% /notice %}}

```

Comment (codinglikeagirl42) [reply content/web-api-rest/reading/rest-fundamentals/_index.md:57]: The above code block to be included in the blue example box.

File: content/web-api-rest/reading/rest-fundamentals/_index.md [hunk 14]
```diff
@@ -0,0 +1,444 @@
...
+```bash {linenos=table}
+[
+    CodingEvent { ... },
+    ...
+]
+```
+{{% /notice %}}
+
+The state of the ``CodingEvent`` collection is made up of the collective state of each ``CodingEvent`` entity within it.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/rest-fundamentals/_index.md:140]:
```suggestion

The state of the ``CodingEvent`` collection is made up of the collective state of each ``CodingEvent`` entity within it.
{{% /notice %}}
```

File: content/web-api-rest/reading/rest-fundamentals/_index.md [hunk 15]
```diff
@@ -0,0 +1,444 @@
...
+
+```bash {linenos=table}
+[
+    Tag { ... },
+    ...
+]
+```
+{{% /notice %}}
+
+A request to the endpoint of the `Tag` collection would include its respective `Tag` entity representations (JSON objects).
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/rest-fundamentals/_index.md:153]:
```suggestion

A request to the endpoint of the `Tag` collection would include its respective `Tag` entity representations (JSON objects).
{{% /notice %}}
```

File: content/web-api-rest/reading/rest-fundamentals/_index.md [hunk 16]
```diff
@@ -0,0 +1,444 @@
...
+- [REST sub-collections, relationships and links](https://restful-api-design.readthedocs.io/en/latest/relationships.html)
+- [OpenAPI specification & Swagger REST tools](https://swagger.io/specification/)
+- [The GitHub API](https://developer.github.com/v3/) and [Stripe (payment processing) API](https://stripe.com/docs/api) are excellent examples of RESTful design (and fantastic documentation)
+
+#### Deep Understanding
+
+- [the REST constraints](https://www.restapitutorial.com/lessons/whatisrest.html)
+- [The Richardson REST maturity model](https://www.martinfowler.com/articles/richardsonMaturityModel.html)
+- [the original REST doctoral thesis by Roy Fielding](https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm)
+
```
Comment (codinglikeagirl42) [inline content/web-api-rest/reading/rest-fundamentals/_index.md:410]: Please move this section to a Next Steps page. Delete Learning More subheading and rework the sentence at line 393 -> something like ->

We have covered the fundamental aspects of the RESTful mental model and practical usage. However, RESTful design is a deep topic that even 
extends beyond the web and use of HTTP!

File: content/web-api-rest/exercises/_index.md [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+Some items to consider:
+
+1. What is the HTTP request type being used?
+1. What is the resource path being requested?
+1. Is there a request body being sent? What is included in it?
+1. If the request is successful, what information can we expect to be included in the response?
+{{% /notice %}}
+
+<!-- TODO: Need proper link here -->
+<!-- :ref:[Check your solutions](web-api-ex-1) -->
```
Comment (codinglikeagirl42) [inline content/web-api-rest/exercises/_index.md:105]:
```suggestion
{{% expand "Check Your Solution" %}}

{{% expand %}}
```

Comment (codinglikeagirl42) [reply content/web-api-rest/exercises/_index.md:105]: See exercises in classes chapter for reference of adding drop down Check your Solution and add solution.

File: content/web-api-rest/studio/_index.md [hunk 18]
```diff
@@ -0,0 +1,359 @@
...
+
+{{% notice blue "Note" "rocket" %}}
+Our focus in this course is on operations and as such we will not be going into the development of the API. However, feel free to explore the source code if you are curious about the similarities and differences between the .NET MVC and API implementations.
+{{% /notice %}}
+
+Let's begin by forking and cloning the repo onto our machine. In your Powershell or terminal window, move into a directory where you plan to save your local copy of the API codebase.
+
+```bash
+git clone https://github.com/<GitUsername>/coding-events-api
+```
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:35]:
```suggestion
```

Comment (codinglikeagirl42) [reply content/web-api-rest/studio/_index.md:35]: Students should be able to do this without prompting, it is the last studio in the second unit.  Suggestion delete.

File: content/web-api-rest/studio/_index.md [hunk 19]
```diff
@@ -0,0 +1,359 @@
...
+```bash
+# cd is an alias (like a nick-name) for the Set-Location cmdlet in PowerShell
+cd coding-events-api
+```
+
+```bash
+# check out the 1-sqlite branch
+git checkout 1-sqlite
+```
+
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:54]:
```suggestion
```

File: content/web-api-rest/studio/_index.md [hunk 20]
```diff
@@ -0,0 +1,359 @@
...
+Keep your terminal open! We have some more work to do.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:55]: Open your terminal and make sure you are in the correct repo and on the correct branch.

Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:55]:
```suggestion
```

File: content/web-api-rest/exercises/_index.md [hunk 21]
```diff
@@ -0,0 +1,215 @@
...
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: 12/15/22 # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+We've created a CodingEvents API application [here](https://github.com/LaunchCodeEducation/coding-events-api/tree/1-sqlite). Before we ask you to fork and clone this, let's first think conceptually about the kinds of requests a client application could make to this API.
+
+## The API
+
+This branch of `coding-events-api` starts by only exposing a single resource, `CodingEvent`, and four endpoints for interacting with it.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/exercises/_index.md:19]:
```suggestion
The 1-sqlite branch of `coding-events-api` starts by only exposing a single resource, `CodingEvent`, and four endpoints for interacting with it.
```

File: content/web-api-rest/studio/_index.md [hunk 22]
```diff
@@ -0,0 +1,359 @@
...
+### List the Coding Events
+
+To create our first request using Postman, select the *Create New* button located within the "Start with something new" section.
+
+![Close up of the Postman create new button](pictures/create-new.png?classes=border)
+
+Select the *Create New* button to create a new request
+
+{{% notice blue "Note" "rocket" %}}
+The above screenshot and the screenshots on this page are of the Linux version of Postman. The *Create New* button on your launchpad should also be located under the "Start with something new" section of the home page.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:117]:
```suggestion
The above screenshot is from the Home page on Postman. The *Create New* button on your launchpad should also be located under the "Start with something new" section.  If you are on the Workshop page choose the *New* button on the top right of the *My Workspace* pane located on the left side.
 ```

File: content/web-api-rest/studio/_index.md [hunk 23]
```diff
@@ -0,0 +1,359 @@
...
+lastMod: 12/15/22 # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+The UI of a browser is designed to make simple `GET` requests for URLs entered into its address bar. This design works great for browsing sites, but falls short when working with APIs. Anything beyond a `GET` request is difficult to send via a browser address bar alone. Think about what is needed to create a new `CodingEvent`. This type of request contains a body. Our MVC application included a view to allow us to test inputs. Our API, however, is **headless**. It does not contain the client-side form. In order to test how it handles requests then, we need a way to interact with the API server without the browser. In this studio, we work with Postman to explore how APIs can be consumed.
+
+## Setup
+
+### Install Postman
+
+If you haven't done so already, [install Postman](postman-installation)
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:21]: Add link to the new installing postman page.

File: content/web-api-rest/studio/_index.md [hunk 24]
```diff
@@ -0,0 +1,359 @@
...
+
+### Fork and Clone the API Source Code
+
+With Postman installed, we're ready to fork and clone the [CodingEvents API](https://github.com/LaunchCodeEducation/coding-events-api/tree/1-sqlite). Although it is implemented differently from CodingEvents MVC, you will find that most of the features from the MVC application have been supported through endpoints in the API.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:25]: This links to the LaunchCode repo - the repo i used was this one ->

https://github.com/jwoolbright23/coding-events-api/tree/1-sqlite

**Review by codinglikeagirl42 — CHANGES_REQUESTED** [review]
Please also add the install postman page to the book. Let me know if you have any questions.

File: content/web-api-rest/reading/rest-fundamentals/_index.md (hunk 16 above)
Comment (jwoolbright23) [reply content/web-api-rest/reading/rest-fundamentals/_index.md:410]: Implemented the above changes.

File: content/web-api-rest/exercises/_index.md (hunk 17 above)
Comment (jwoolbright23) [reply content/web-api-rest/exercises/_index.md]: I do not think we need to add a solution to the exercises on this page after looking into it a bit further. The answers to the questions we ask are explained within the question itself. This is more of a check your understanding similar to a reading walkthrough.

File: content/web-api-rest/exercises/_index.md (hunk 21 above)
Comment (jwoolbright23) [reply content/web-api-rest/exercises/_index.md:19]: Resolved this by removing the 1-sqlite branch altogether and having the learners work off of main.

File: content/web-api-rest/studio/_index.md (hunk 24 above)
Comment (jwoolbright23) [reply content/web-api-rest/studio/_index.md]: resolved link issues here

File: content/web-api-rest/studio/_index.md (hunk 23 above)
Comment (jwoolbright23) [reply content/web-api-rest/studio/_index.md]: New page added and linked to.

File: content/web-api-rest/exercises/_index.md (hunk 17 above)
Comment (codinglikeagirl42) [reply content/web-api-rest/exercises/_index.md]: Thanks for the explanation.

File: content/appendices/install-postman/_index.md [hunk 25]
```diff
@@ -0,0 +1,57 @@
+---
+title: "Installing Postman"
+date: 2023-01-20T08:49:09-06:00
+draft: false
+weight: 75
+originalAuthor: John Woolbright # to be set by page creator
+originalAuthorGitHub: jwoolbright23 # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
```
Comment (codinglikeagirl42) [inline content/appendices/install-postman/_index.md:9]:
```suggestion
reviewer: Kimberly Horan # to be set by the page reviewer
reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
```

File: content/web-api-rest/studio/_index.md [hunk 26]
```diff
@@ -0,0 +1,327 @@
...
+lastMod: 12/15/22 # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+The UI of a browser is designed to make simple `GET` requests for URLs entered into its address bar. This design works great for browsing sites, but falls short when working with APIs. Anything beyond a `GET` request is difficult to send via a browser address bar alone. Think about what is needed to create a new `CodingEvent`. This type of request contains a body. Our MVC application included a view to allow us to test inputs. Our API, however, is **headless**. It does not contain the client-side form. In order to test how it handles requests then, we need a way to interact with the API server without the browser. In this studio, we work with Postman to explore how APIs can be consumed.
+
+## Setup
+
+### Install Postman
+
+If you haven't done so already, [install Postman](https://education.launchcode.org/csharp-web-dev-curriculum/appendices/install-postman/)
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:21]:
```suggestion
If you haven't done so already, [install Postman]({{< relref "../../../appendices/install-postman/_index.md" >}}).
```

File: content/web-api-rest/studio/_index.md [hunk 27]
```diff
@@ -0,0 +1,327 @@
...
+
+Let's change into the repo and switch to this branch:
+
+```bash
+# cd is an alias (like a nick-name) for the Set-Location cmdlet in PowerShell
+cd coding-events-api
+```
+
+Keep your terminal open! We have some more work to do.
+
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:45]:
```suggestion
```

File: content/web-api-rest/studio/_index.md [hunk 28]
```diff
@@ -0,0 +1,327 @@
...
+### Start the API Server
+
+We'll start the API server from the terminal using the `dotnet run` command. Navigate to the `CodingEventsAPI` project folder *within* your `coding-events-api` solution. This is the folder that contains `Controllers/` and so on, and is NOT the main project folder.
+
+```bash
+# change to the CodingEventsAPI project directory
+cd CodingEventsAPI
+```
```
Comment (codinglikeagirl42) [inline content/web-api-rest/studio/_index.md:53]:
```suggestion
```bash
# navigate and change into the coding-events-api solution 
cd coding-events-api
# change to the CodingEventsAPI project directory
cd CodingEventsAPI
```
```

File: content/web-api-rest/exercises/_index.md [hunk 29]
```diff
@@ -0,0 +1,210 @@
...
+1. What is the resource path being requested?
+1. Is there a request body being sent? What is included in it?
+1. If the request is successful, what information can we expect to be included in the response?
+1. If the request contains an error, what information can we expect to be included in the response?
+{{% /notice %}}
+
+### Install Postman
+
+Now that we've explored working with those endpoints, we're almost ready to start running the API and test sending those requests. 
+You'll need to [install Postman](https://education.launchcode.org/csharp-web-dev-curriculum/appendices/install-postman/) to work with this lesson's studio and practice running these requests.
```
Comment (codinglikeagirl42) [inline content/web-api-rest/exercises/_index.md:209]:
```suggestion
You'll need to [install Postman]({{< relref "../../../appendices/install-postman/_index.md" >}}) to work with this lesson's studio and practice running these requests.
```

File: content/web-api-rest/exercises/_index.md [hunk 30]
```diff
@@ -0,0 +1,210 @@
...
+conversion, or serialization, so that our API can output data in a portable format that is language-agnostic.
+
+### The Endpoints
+
+This branch of the API has four endpoints: 
+
+1. [GET Coding Events](https://education.launchcode.org/csharp-web-dev-curriculum/web-api-rest/exercises/#get-coding-events)
+1. [GET a Single Coding Event](https://education.launchcode.org/csharp-web-dev-curriculum/web-api-rest/exercises/#get-a-single-coding-event)
+1. [Create a Coding Event](https://education.launchcode.org/csharp-web-dev-curriculum/web-api-rest/exercises/#create-a-coding-event)
+1. [Delete a Coding Event](https://education.launchcode.org/csharp-web-dev-curriculum/web-api-rest/exercises/#delete-a-coding-event)
```
Comment (codinglikeagirl42) [inline content/web-api-rest/exercises/_index.md:68]:
```suggestion
1. [GET Coding Events](#get-coding-events)
1. [GET a Single Coding Event](#get-a-single-coding-event)
1. [Create a Coding Event](#create-a-coding-event)
1. [Delete a Coding Event](#delete-a-coding-event)
```

**Review by codinglikeagirl42 — APPROVED** [review]
Please add *.swp to .gitignore and remove from the repo -> appendices/install-postman/._index.md.swp

Then merge away!

**Pull request merged**